### PR TITLE
move getServiceId function into /utils

### PR DESCRIPTION
**Pull request by gestchild**

### CODE REVIEW

File: common/utils/iiif.js [hunk 1]
```diff
@@ -7,6 +7,23 @@ import {
   type IIIFMediaElement,
 } from '../model/iiif';
 
+export function getServiceId(currentCanvas: ?IIIFCanvas) {
+  if (!currentCanvas) return null;
+  const serviceSrc = currentCanvas.images[0].resource.service;
```
Comment (jamesgorrie) [inline common/utils/iiif.js:12]: There's always a `image`?

File: common/utils/iiif.js [hunk 2]
```diff
@@ -7,6 +7,23 @@ import {
...
+        item => item['@context'] === 'http://iiif.io/api/image/2/context.json'
+      );
+      return service && service['@id'];
+    } else {
+      return serviceSrc['@id'];
+    }
+  } else {
+    return null;
+  }
+}
```
Comment (jamesgorrie) [inline common/utils/iiif.js:25]: Might be worth seeing if we can get the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) in for this stuff?

File: common/utils/iiif.js [hunk 3]
```diff
@@ -8,8 +8,7 @@ import {
 } from '../model/iiif';
 
 export function getServiceId(currentCanvas: ?IIIFCanvas) {
-  if (!currentCanvas) return null;
-  const serviceSrc = currentCanvas.images[0].resource.service;
+  const serviceSrc = currentCanvas?.images[0]?.resource?.service;
```
Comment (davidpmccormick) [inline common/utils/iiif.js:11]: Oh nice – I missed this going in.

Comment (jamesgorrie) [reply common/utils/iiif.js:11]: @gestchild you could get it in for TS too, or wait till we use it there.

File: common/utils/iiif.js [hunk 4]
```diff
@@ -7,6 +7,22 @@ import {
   type IIIFMediaElement,
 } from '../model/iiif';
 
+export function getServiceId(currentCanvas: ?IIIFCanvas) {
```
Comment (jamesgorrie) [inline common/utils/iiif.js:10]: What's the return type for this?

Comment (jamesgorrie) [reply common/utils/iiif.js:10]: It might be good to see if we can lint on not using types.

**Pull request merged**